File: src/main/java/stni/atlassian/remote/queryold/QueryService.java
package stni.atlassian.remote.queryold;

import com.atlassian.jira.rpc.soap.beans.RemoteIssue;
import com.atlassian.jira.rpc.soap.beans.RemoteProject;

import java.util.List;

/**
 *
 */
public interface QueryService {
    List<RemoteProject> getProjectsByKey(String... keys);

    RemoteIssue getIssue(String issueKey);

    RemoteIssue[] getIssuesFromFilter(String filter);

    RemoteIssue[] getIssuesFromJqlSearch(String query, int maxResults);

    String getBaseUrl();

    String getCustomField(RemoteIssue issue, String name);
}

File: src/main/scala/guru/nidi/atlassian/remote/query/QueryService.scala
package guru.nidi.atlassian.remote.query

import com.atlassian.jira.rpc.soap.beans._
import guru.nidi.atlassian.remote.jira.RemoteIssueExt

/**
 *
 */
trait QueryService {
  def getProjectsByKey(keys: String*): Seq[RemoteProject]

  def getIssue(issueKey: String): RemoteIssueExt

  def getIssuesFromJqlSearch(query: String, maxResults: Int): Seq[RemoteIssueExt]

  def baseUrl: String

  def customField(issue: RemoteIssue, name: String): String

  def priorityById(id: String): RemotePriority

  def issueTypeById(id: String): RemoteIssueType

  def statusById(id: String): RemoteStatus
  
  def resolutionById(id: String): RemoteResolution
}
